File: round-1a/01-append/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>

#define REP(i, n) for (int i = 0; i < (n); i++)

using namespace std;

int main() {
  int T;
  cin >> T;
  REP(t, T) {
    int N;
    cin >> N;
    vector<string> X(N);
    REP(i, N) cin >> X[i];
    int ans = 0;
    string prev = "";
    // int prev = 0;
    REP(i, N) {
      string x = X[i];
      if (prev.size() < x.size()) {
        prev = x;
        continue;
      } else if (prev.size() == x.size()) {
        if (stoll(prev) < stoll(x)) {
          prev = x;
          continue;
        } else {
          prev = x + "0";
          ans += 1;
          continue;
        }
      } else {
        REP(j, x.size()) {
          if (prev[j] > x[j]) {
            int add = prev.size() - x.size() + 1;
            ans += add;
            prev = x;
            REP(_, add) prev.push_back('0');
            goto done;
          }
          if (prev[j] < x[j]) {
            int add = prev.size() - x.size();
            ans += add;
            prev = x;
            REP(_, add) prev.push_back('0');
            goto done;
          }
        }
        {
          string prev_inc = to_string(stoll(prev) + 1);
          if (x == prev_inc.substr(0, x.size())) {
            int add = prev.size() - x.size();
            ans += add;
            prev = prev_inc;
            goto done;
          } else {
            int add = prev.size() - x.size() + 1;
            ans += add;
            prev = x;
            REP(_, add) prev.push_back('0');
            goto done;
          }
        }
        done:;
      }
    }
    cout << "Case #" << (t + 1) << ": " << ans << "\n";
  }
}

File: round-1a/01-append/main.py
T = int(input())

for t in range(T):
  N = int(input())
  X = input().split()
  prev = ""
  ans = 0
  for x in X:
    if len(prev) < len(x):
      prev = x
      continue
    elif len(prev) == len(x):
      if int(prev) < int(x):
        prev = x
        continue
      else:
        prev = x + "0"
        ans += 1
        continue
    else:
      y = prev[0 : len(x)]
      if y > x:
        add = len(prev) - len(x) + 1
        ans += add
        prev = x + "0" * add
        continue
      elif y < x:
        add = len(prev) - len(x)
        ans += add
        prev = x + "0" * add
        continue
      else:
        prev_inc = str(int(prev) + 1)
        if x == prev_inc[0 : len(x)]:
          add = len(prev) - len(x)
          ans += add
          prev = prev_inc
          continue
        else:
          add = len(prev) - len(x) + 1
          ans += add
          prev = x + "0" * add
          continue
  print(f"Case #{t + 1}: {ans}")
